nullaway/src/main/java/com/uber/nullaway/handlers/OptionalEmptinessHandler.java

```diff
@@ -51,9 +62,13 @@
 public class OptionalEmptinessHandler extends BaseNoOpHandler {
 
   @Nullable private ImmutableSet<Type> optionalTypes;
+  private NullAway analysis;
```

Comment:
Let's just pass this one on the handler method signature (either `onExpressionDereference` or `checkErrorMessageInDereference` or whatever, rather than holding on to the reference to the core analysis in a field here.

Reply:
I did that first but then changed it after feedback. I think in our current code we are doing it in both ways at different places. Let's make it consistent later at all places?

Reply:
I think I gave the feedback to switch it to this way, based on some other handler.  The thing is we already pass `NullAway` to the top-level class handler.  I agree we can clean this up later; @lazaroclapp?

Reply:
I think if we rename the other method to `onExpressionDereference` it should take the NullAway instance there for consistency with other `Handler.onX` methods. Given that, while I don't terribly mind keeping the NullAway instance around in and of itself, I also don't see why we need to if the specific method will take it too :)

Reply:
Actually, on second thought, we already have a bunch of `onDataflow...` methods that don't take a `NullAway` parameter. I am fine with leaving this as in the latest commit (3d59615)